tests/e2e/framework/rawVM.go

```diff
@@ -0,0 +1,240 @@
+type RawVM interface {
+	Cleanable
+	GetInternalIP() (string, error)
+	GetExternalIP() (string, error)
+	SecureShell(cmd string) (string, error)
+	SecureCopy(files ...string) (string, error)
+}
+
+var (
+	// flags to select vm with specific configuration
+	projectID     = flag.String("project_id", "istio-testing", "Project ID")
+	projectNumber = flag.String("project_number", "450874614208", "Project Number")
+	vmZone        = flag.String("vm_zone", "us-west1-a", "The zone in which the VM resides")
+	clusterName   = flag.String("cluster_name", "istio-e2e-rbac-rotation-1", "The name of the istio cluster that the VM extends")
+	clusterZone   = flag.String("cluster_zone", "us-east4-c", "The zone of the istio cluster that the VM extends")
+	image         = flag.String("image", "debian-9-stretch-v20170816", "Image Name")
+	imageProject  = flag.String("image_project", "debian-cloud", "Image Project")
+	// paths
+	setupMeshExScript  = ""
+	mashExpansionYaml  = ""
+	setupIstioVMScript = ""
+	// kubectl delete resources on tear down
+	kubeDeleteYaml = []string{}
+)
+
+// GCPRawVM is hosted on Google Cloud Platform
+type GCPRawVM struct {
+	Name        string
+	ClusterName string
+	ClusterZone string
+	Namespace   string
+	// ServiceAccount must have iam.serviceAccountActor or owner permissions
+	// to the project. Use IAM settings.
+	ServiceAccount string
+	ProjectID      string
+	Zone           string
+	Image          string
+	ImageProject   string
+}
+
+// NewGCPRawVM creates a new vm on GCP
+func NewGCPRawVM(namespace string) *GCPRawVM {
+	return &GCPRawVM{
+		Name:           "raw-vm-test-new", // TODO (chx) generate by vm pool server
+		ClusterName:    *clusterName,
+		ClusterZone:    *clusterZone,
+		Namespace:      namespace,
+		ServiceAccount: fmt.Sprintf("%s-compute@developer.gserviceaccount.com", *projectNumber), //"istio-prow-test-job@istio-testing.iam.gserviceaccount.com"
+		ProjectID:      *projectID,
+		Zone:           *vmZone,
+		Image:          *image,
+		ImageProject:   *imageProject,
+	}
+}
+
+// GetInternalIP returns the internal IP of the VM
+func (vm *GCPRawVM) GetInternalIP() (string, error) {
+	cmd := vm.baseCommand("describe") +
+		" --format='value(networkInterfaces[0].accessConfigs[0].natIP)'"
+	o, e := u.Shell(cmd)
+	return strings.Trim(o, "\n"), e
+}
+
+// GetExternalIP returns the internal IP of the VM
+func (vm *GCPRawVM) GetExternalIP() (string, error) {
+	cmd := vm.baseCommand("describe") +
+		" --format='value(networkInterfaces[0].networkIP)'"
+	o, e := u.Shell(cmd)
+	return strings.Trim(o, "\n"), e
+}
+
+// SecureShell execeutes cmd on vm through ssh
+func (vm *GCPRawVM) SecureShell(cmd string) (string, error) {
+	ssh := fmt.Sprintf("gcloud compute ssh --project %s --zone %s %s --command \"%s\"",
+		vm.ProjectID, vm.Zone, vm.Name, cmd)
+	return u.Shell(ssh)
+}
+
+// SecureCopy copies files to vm via scp
+func (vm *GCPRawVM) SecureCopy(files ...string) (string, error) {
```

Comment:
this isn't used right ?

Reply:
It is not used at the moment, but it is useful when dev adds more tests to test the VM.